src/posix/platform/radio_spinel.cpp

```diff
@@ -452,6 +478,157 @@ void RadioSpinel::HandleResponse(const uint8_t *aBuffer, uint16_t aLength)
+            VerifyOrExit(len == OT_EXT_PAN_ID_SIZE, error = OT_ERROR_INVALID_ARGS);
+            memcpy(dataset.mExtendedPanId.m8, xpanid, len);
+            dataset.mComponents.mIsExtendedPanIdPresent = true;
+            break;
+        }
+
+        case SPINEL_PROP_IPV6_ML_PREFIX:
+        {
+            const otIp6Address *addr;
+            uint8_t             prefixLen;
+
+            SuccessOrExit(error = decoder.ReadIp6Address(addr));
+            SuccessOrExit(error = decoder.ReadUint8(prefixLen));
+            VerifyOrExit(prefixLen == 64, error = OT_ERROR_INVALID_ARGS);
+            memcpy(dataset.mMeshLocalPrefix.m8, addr, OT_MESH_LOCAL_PREFIX_SIZE);
+            dataset.mComponents.mIsMeshLocalPrefixPresent = true;
+            break;
+        }
+
+        case SPINEL_PROP_DATASET_DELAY_TIMER:
+        {
+            SuccessOrExit(error = decoder.ReadUint32(dataset.mDelay));
+            dataset.mComponents.mIsDelayPresent = true;
+            break;
+        }
+
+        case SPINEL_PROP_MAC_15_4_PANID:
+        {
+            SuccessOrExit(error = decoder.ReadUint16(dataset.mPanId));
+            dataset.mComponents.mIsPanIdPresent = true;
+            break;
+        }
+
+        case SPINEL_PROP_PHY_CHAN:
+        {
+            uint8_t channel;
+
+            SuccessOrExit(error = decoder.ReadUint8(channel));
+            dataset.mChannel                      = channel;
+            dataset.mComponents.mIsChannelPresent = true;
+            break;
+        }
+
+        case SPINEL_PROP_NET_PSKC:
+        {
+            const uint8_t *psk;
+            uint16_t       len;
+
+            SuccessOrExit(error = decoder.ReadData(psk, len));
+            VerifyOrExit(len == OT_PSKC_MAX_SIZE, error = OT_ERROR_INVALID_ARGS);
+            memcpy(dataset.mPskc.m8, psk, OT_PSKC_MAX_SIZE);
+            dataset.mComponents.mIsPskcPresent = true;
+            break;
+        }
+
+        case SPINEL_PROP_DATASET_SECURITY_POLICY:
+        {
+            SuccessOrExit(error = decoder.ReadUint16(dataset.mSecurityPolicy.mRotationTime));
+            SuccessOrExit(error = decoder.ReadUint8(dataset.mSecurityPolicy.mFlags));
+            dataset.mComponents.mIsSecurityPolicyPresent = true;
+            break;
+        }
+
+        case SPINEL_PROP_PHY_CHAN_SUPPORTED:
+        {
+            uint8_t channel;
+
+            dataset.mChannelMask = 0;
+
+            while (!decoder.IsAllReadInStruct())
+            {
+                SuccessOrExit(error = decoder.ReadUint8(channel));
+                VerifyOrExit(channel <= 31, error = OT_ERROR_INVALID_ARGS);
+                dataset.mChannelMask |= (1UL << channel);
+            }
+            dataset.mComponents.mIsChannelMaskPresent = true;
+            break;
+        }
+
+        default:
+            break;
+        }
+
+        SuccessOrExit(error = decoder.CloseStruct());
+    }
+
+    dataset.mActiveTimestamp                      = 0;
+    dataset.mComponents.mIsActiveTimestampPresent = true;
```

Comment:
"We may want to consider initially saving with Active Timestamp value set to 0. This will allow us to still join the network yet retrieve the full Active Dataset from a neighboring device if one exists."

I will add a comment in code to explain this.

Reply:
Do we always want to retrieve it from neighbor again? If we already have it and if it has a higher time stamp than neighbor's, I think we should use ours.

Another point is about pending dataset, I think we need pending time stamp to be set for it?

Reply:
Spinel does not currently allow retrieving the full Operational Dataset (e.g. attributes that are not supported by spinel itself). If we updated spinel (and this solution) to retrieve the Operational Dataset as a binary blob, then I agree we should just keep the Operational Dataset as-is, including the timestamp.

Reply:
Got it. Thanks. Makes sense as a solution for now.